tree/ntuple/v7/src/RField.cxx

```diff
@@ -2006,6 +2006,11 @@ void ROOT::Experimental::RRVecField::ReadGlobalImpl(NTupleSize_t globalIndex, vo
          CallGenerateValueOn(*fSubFields[0], begin + (i * fItemSize));
    }
 
+   if (fSubFields[0]->IsSimple()) {
+      GetPrincipalColumnOf(*fSubFields[0])->ReadV(collectionStart, nItems, begin);
+      return;
```

Comment:
The change definitely makes sense. Unfortunately, I can't tell how this leads to the current failures in the test `onDiskPage && (bytesOnStorage == onDiskPage->GetSize()) violated at line 341`

Reply:
Found it: empty simple vectors trigger that. Should be fixed now.